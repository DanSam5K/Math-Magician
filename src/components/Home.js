import React from 'react';

const Home = () => (
  <div>
    <h2 className="home-title">Welcome to Magician Page!</h2>
    <p className="paragraph">
      Math magicians is a website for all fans of mathematics.
      <br />
      It is a Single Page App (SPA) that allows users to:
      <ul>
        <li>Make simple calculations.</li>
        <li>Read a random math-related quote.</li>
      </ul>
    </p>
    <p className="paragraph">By building this simple application, I was able to learn how to use React.</p>
  </div>
);

export default Home;
