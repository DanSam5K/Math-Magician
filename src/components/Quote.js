import React from 'react';

const Quote = () => (
  <p className="p-quote">Mathematics is not about numbers, equations, computations, or algorithms: It`s about understanding. - Williams Paul</p>
);

export default Quote;
