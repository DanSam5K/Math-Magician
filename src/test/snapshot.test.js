import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../components/Home';
import Quote from '../components/Quote';



describe('Components Snapshots test', () => {

	it('renders home as intended', () => {
	   const UIHome= renderer
		 .create(<Home />)
		 .toJSON();
		 expect(UIHome).toMatchSnapshot();
	});

	it('renders quote as intended', () => {
	   const UIQuote = renderer
		 .create(<Quote />)
		 .toJSON();
		 expect(UIQuote).toMatchSnapshot();
	});

})
