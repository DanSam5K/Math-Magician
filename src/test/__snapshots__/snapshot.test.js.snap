// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Snapshots test renders home as intended 1`] = `
<div>
  <h2
    className="home-title"
  >
    Welcome to Magician Page!
  </h2>
  <p
    className="paragraph"
  >
    Math magicians is a website for all fans of mathematics.
    <br />
    It is a Single Page App (SPA) that allows users to:
    <ul>
      <li>
        Make simple calculations.
      </li>
      <li>
        Read a random math-related quote.
      </li>
    </ul>
  </p>
  <p
    className="paragraph"
  >
    By building this simple application, I was able to learn how to use React.
  </p>
</div>
`;

exports[`Components Snapshots test renders quote as intended 1`] = `
<p
  className="p-quote"
>
  Mathematics is not about numbers, equations, computations, or algorithms: It\`s about understanding. - Williams Paul
</p>
`;
